### ISSUE TITLE
Block system.io for Lean

### ISSUE THREAD

**Issue by swamp-agr** [swamp-agr/proof-assistant-bot#23]
Command example: 

```lean
/lean import system.io
#eval io.proc.spawn { cmd := "/usr/bin/ls" }
```

**Comment by swamp-agr** [comment]
While dirty hack was good and worked, the `system.io` module content could be pasted directly and used in a following way:

```lean
/lean inductive io.error
| other     : string → io.error
| sys       : nat → io.error
inductive io.mode
| read | write | read_write | append
inductive io.process.stdio
| piped
| inherit
| null
structure io.process.spawn_args :=
(cmd : string)
(args : list string := [])
(stdin := stdio.inherit)
(stdout := stdio.inherit)
(stderr := stdio.inherit)
(cwd : option string := none)
(env : list (string × option string) := [])

class monad_io (m : Type → Type → Type) :=
[monad    : Π e, monad (m e)]
(catch    : Π e₁ e₂ α, m e₁ α → (e₁ → m e₂ α) → m e₂ α)
(fail     : Π e α, e → m e α)
(iterate  : Π e α, α → (α → m e (option α)) → m e α)
(handle   : Type)
open monad_io (handle)
class monad_io_process (m : Type → Type → Type) [monad_io m] :=
(child  : Type)
(stdin  : child → (handle m))
(stdout : child → (handle m))
(stderr : child → (handle m))
(spawn  : io.process.spawn_args → m io.error child)
(wait   : child → m io.error nat)
(sleep  : nat → m io.error unit)
constant io_core : Type → Type → Type
@[instance] constant monad_io_impl         : monad_io io_core
@[instance] constant monad_io_process_impl : monad_io_process io_core
@[reducible] def io (α : Type) := io_core io.error α
def child : Type := monad_io_process.child io_core
def spawn (p : io.process.spawn_args) : io child := monad_io_process.spawn p
#eval spawn {cmd := "/usr/bin/ls"}
```

**Comment by swamp-agr** [comment]
Closed via #31.
